include/host/wasi_crypto/utils/secret_key.h

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2019-2022 Second State INC
+
+//===-- wasi_crypto/utils/secret_key.h - Secret key definition ------===//
+//
+// Part of the WasmEdge Project.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains definition of secret key
+///
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "host/wasi_crypto/utils/error.h"
+#include "common/span.h"
+#include "openssl/rand.h"
+
+#include <climits>
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+namespace WasmEdge {
+namespace Host {
+namespace WasiCrypto {
+
+/// A vector wrapper, but swipe secret key info on destory
+class SecretKey {
+public:
+  SecretKey(std::vector<uint8_t> Data) : Data(std::move(Data)) {}
+  
+  SecretKey(Span<const uint8_t> Data) : Data(Data.begin(), Data.end()) {}
+
+  SecretKey(size_t Size) : Data(Size) {}
+
+  ~SecretKey() { std::fill(Data.begin(), Data.end(), 0); }
```

Comment:
If you want to clean all the data in vector to `0`, use `OPENSSL_cleanse` to prevent compiler optimization to remove this action.